pkg/master/master_test.go

```diff
@@ -430,6 +432,60 @@ func TestGenerateSSHKey(t *testing.T) {
 	// TODO: testing error cases where the file can not be removed?
 }
 
+func TestDiscoveryAtAPIS(t *testing.T) {
+	master, config, assert := setUp(t)
+	master.exp = true
+	// ================= preparation for master.init() ======================
```

Comment:
@lavalamp this test needs your review. I set many default values to make master.init to execute without error. Please take a look if these default values make sense. Thank you.